diff --git a/cpp/tests/unit/CMakeLists.txt b/cpp/tests/unit/CMakeLists.txt
index ee2d28ddb..fdcda220d 100644
--- a/cpp/tests/unit/CMakeLists.txt
+++ b/cpp/tests/unit/CMakeLists.txt
@@ -1,5 +1,6 @@
 set(unittests_headers    
 
+        ./utils/DecoyGateCommandHandler.h
     ./utils/APDUHelpers.h
     ./utils/APDUHexBuilders.h
     ./utils/BufferHelpers.h
@@ -20,6 +21,9 @@ set(unittests_headers
 
 set(unittests_src
     ./main.cpp
+    ./TestDecoyGateEndpoint.cpp
+    ./TestDecoyGateMasterAcceptance.cpp
+    ./TestDecoyGateReadSize.cpp
 
     ./TestAPDUParsing.cpp
     ./TestAPDUWriting.cpp    
